impl/src/main/java/org/ehcache/config/builders/CacheConfigurationBuilder.java

```diff
@@ -667,4 +628,22 @@ private void removeExistingSerializerConfigFor(DefaultSerializerConfiguration.Ty
       serviceConfigurations.toArray(new ServiceConfiguration<?>[serviceConfigurations.size()]));
 
   }
+
+  private <T extends ServiceConfiguration<?>> CacheConfigurationBuilder<K,V> addOrReplaceConfiguration(T configuration) {
+    return addOrReplaceConfiguration((Class<T>) configuration.getClass(), configuration);
+  }
+
+  private <T extends ServiceConfiguration<?>> CacheConfigurationBuilder<K,V> addOrReplaceConfiguration(Class<T> configurationType, T configuration) {
+    return mapServiceConfiguration(configurationType, e -> configuration);
+  }
+
+  private <T extends ServiceConfiguration<?>> CacheConfigurationBuilder<K,V> mapServiceConfiguration(Class<T> configurationType, Function<Optional<T>, T> mapper) {
+    CacheConfigurationBuilder<K, V> otherBuilder = new CacheConfigurationBuilder<>(this);
+    T existingServiceConfiguration = otherBuilder.getExistingServiceConfiguration(configurationType);
+    if (existingServiceConfiguration != null) {
+      otherBuilder.serviceConfigurations.remove(existingServiceConfiguration);
+    }
+    otherBuilder.serviceConfigurations.add(mapper.apply(Optional.ofNullable(existingServiceConfiguration)));
```

Comment:
Passing an optional in parameter isn't considered a good idea

Reply:
Okay. Will move to a `UnaryOperator<T>`